### DISCUSSION TITLE
The future of Microsoft desktop development?

### DISCUSSION THREAD

**Discussion by MarkBrainfire** [microsoft/microsoft-ui-xaml#8243] [General]
![WinUI3_Languishing](https://user-images.githubusercontent.com/78389339/222561704-97ba4f4f-48fc-47f3-8e55-09a4f988595b.jpg)

**Comment by oliverw** [comment]
This wouldn't be concerning if the technology would have just become mature over time but given the > 2k open issues, it isn't looking good. First we got screwed over by WPF, then Silverlight, Windows Phone, UWP and finally WinUI. The sudden release of a [WPF 2023 roadmap](https://github.com/dotnet/wpf/blob/main/roadmap.md), though very welcome, certainly makes the situation even more confusing.

**Reply by MarkBrainfire** [reply]
Don't forget VB and Silverlight.

**Reply by oliverw** [reply]
Right.

**Reply by pjmlp** [reply]
XNA as well.

**Comment by MarkBrainfire** [comment]
Microsoft is squandering an opportunity to define a new approach for software development.  By combining WinUI 3 with Uno Platform it is finally possible to create a no-compromise desktop application that can run locally at the fastest speeds without requiring the Internet and to run natively on any kind of desktop: Windows, MacOS, and Linux.  And that same desktop application can also run natively on Android and iOS mobile devices, and in the browser via WASM.  

Instead, Microsoft is focused on MAUI and Blazor.  Both great technologies, but I would never choose either of these technologies to create an intricate desktop application.  I would enthusiastically embrace C#, .NET and WinUI 3 to create a desktop application and then take that application EVERYWHERE with the Uno Platform.

This is what I'm promoting at https://WinUI3.org

**Comment by omikhailov** [comment]
To be fair, this is a repository for a controls library that is already stable and hasn't changed much in recent years. Therefore, it is not surprising that activity in this particular repository is declining. Right now, in 2023, it is easier to build a WinUI application than a WPF application, because the latter is almost useless without third party libraries, while in WinUI you have almost all controls with Fluent styles out of the box (except the DataGrid, unfortunatelly). The difficulties are more related to SDK and tools that are not open sourced and therefore we cannot correctly assess how actively they are being developed. But I agree that it would be nice to see a more ambitious strategy and healthier callobaration between various teams within Microsoft.

**Reply by pjmlp** [reply]
We can easily access how they aren't being developed.

No designer, .NET still cannot AOT compile like .NET Native, C++/CX was deprecated in 2016 and C++/WinRT is still a joke on us regarding its Visual C++ 6.0 like development experience for COM authoring (as if the time stood still for some folks at WinDev).

**Comment by sylveon** [comment]
This repo is for WinUI 2, a controls library for UWP, and not WinUI 3, a desktop application framework. WinUI 3 issues are taken here, but the code is for WinUI 2 (WinUI 3 is not open source yet), so the commit graph represents WinUI 2 commits. It only makes sense that WinUI 2 languishes and gets less commit activity as WinUI 3 gets more investment.

This is unfortunate for me as I cannot yet migrate to WinUI 3 and WinUI 2 bugs aren't getting fixed anymore :(

**Reply by MarkBrainfire** [reply]
What prevents you from moving to WinUI 3 ?

**Reply by sylveon** [reply]
Out of bounds flyouts, primarily

**Comment by pjmlp** [comment]
After being burnt with downgrade in development experience versus .NET Native and C++/CX, it is clear that the future is the past, Win32, Forms , MFC.

Heck, even C++/WinRT feels like using ATL than anything else.

I have moved into Azure OS with Web UIs.
